src/SmartTransactionsController.ts

```diff
@@ -252,8 +289,10 @@ export default class SmartTransactionsController extends BaseController<
     return currentIndex === -1 || currentIndex === undefined;
   }
 
-  updateSmartTransaction(smartTransaction: SmartTransaction): void {
-    const { chainId } = this.config;
+  updateSmartTransaction(
+    smartTransaction: SmartTransaction,
+    chainId: Hex,
```

Comment:
This is a breaking change — necessary?

Reply:
So in this methods case its kindof an annoying situation with this being a public method (it is called directly from the extension) and one used a bunch internally, where a slightly different method signature can be used to avoid a bunch of duplicate calls to `getNetworkByClientId`. So I think my preferred approach here would be to create a wrapper public method that calls a private version of this method and just takes `smartTransactions` and `networkClientId` which we should expect the client to have in the future.

Reply:
this commit is what I'm thinking for this: https://github.com/MetaMask/smart-transactions-controller/pull/210/commits/684bc5923e2e105473626208e2b1ebe71e181928

Reply:
I'll do another review round later so I can see it in context, but your change makes sense I believe!